storagenode/storagenodedb/pieceinfo.go

```diff
@@ -47,6 +47,31 @@ func (db *pieceinfo) Add(ctx context.Context, info *pieces.Info) (err error) {
 	return ErrInfo.Wrap(err)
 }
 
+// GetAll gets piece information by satellite id.
+func (db *pieceinfo) GetAll(ctx context.Context, satelliteID storj.NodeID) (infos []*pieces.Info, err error) {
```

Comment:
Instead of this function, it would be better to have `GetPage` function that paginates over the database.

It should take three additional params: `createdBefore time.Time` and `offset, limit int`.

Offset and limit should be use to return paginated result from the DB as described here: https://www.petefreitag.com/item/451.cfm